feat(openai): `minimal` and `verbosity`

mdrxy opened the pull request.

mdrxy committed:
.

mdrxy committed:
Merge branch 'master' into mdrxy/openai-verbosity

ccurme commented on libs/partners/openai/tests/integration_tests/chat_models/test_base.py:
```diff
@@ -418,7 +418,7 @@ class MakeASandwich(BaseModel):
 
 def test_tool_use() -> None:
-    llm = ChatOpenAI(model="gpt-4-turbo", temperature=0)
+    llm = ChatOpenAI(model="gpt-5", temperature=0)
```
do the tests pass with gpt-5-nano?

mdrxy committed:
Merge branch 'master' into mdrxy/openai-verbosity

mdrxy committed:
.

mdrxy committed:
.

ccurme commented on libs/partners/openai/tests/integration_tests/chat_models/test_base.py:
```diff
@@ -875,7 +871,7 @@ def test_audio_output_modality() -> None:
 def test_audio_input_modality() -> None:
     llm = ChatOpenAI(
-        model="gpt-4o-audio-preview",
+        model="gpt-5",
```
does this work?

mdrxy committed:
.

mdrxy committed:
.

mdrxy committed:
.

mdrxy committed:
.

ccurme commented on libs/partners/openai/tests/integration_tests/chat_models/test_base.py:
```diff
@@ -1041,8 +1041,32 @@ def test_o1_stream_default_works() -> None:
+    response = ChatOpenAI(
+        model="gpt-5",
+        reasoning_effort="minimal",
+        use_responses_api=use_responses_api,
```
```suggestion
        output_version="responses/v1",
```
(shouldn't make a difference here but we should just use this everywhere for the responses API)

ccurme commented on libs/partners/openai/langchain_openai/chat_models/base.py:
```diff
@@ -1178,6 +1210,13 @@ def _get_request_payload(
         payload = {**self._default_params, **kwargs}
+
+        if self.model_name.startswith("gpt-5"):
+            # gpt-5 models don't support these parameters
```
(here and above) I think it's OK to let openai raise BadRequestError or whatever it is instead of silently dropping these.

I don't think cataloging what models maintain what features and warning / dropping parameters is worthwhile— it's also not forwards compatible if support is added later.

Temperature may be an exception since it's in the LS playground and is common. Actually ChatOpenAI had a default temperature of 0.7 in langchain-openai < 0.3, so `o1` broke out of the box, that's why we had the fix here.

ccurme commented on libs/partners/openai/langchain_openai/chat_models/base.py:
```diff
@@ -527,6 +527,17 @@ class BaseChatOpenAI(BaseChatModel):
+    """
+    verbosity: Optional[str] = None
+    """Controls the verbosity level of responses for reasoning models. For use with the
+    Responses API.
```
does this work in the Responses API?

mdrxy replied:
works with some reasoning models (o1) in regular Chat API but gets rejected by Responses API

will add TODO test and xfail for now

ccurme replied:
docstring should probably not specify "For use with the Responses API" in the meantime.

ccurme reviewed (approved):
Looks good although not sure we should be dropping params (audio, modalities, etc). BadRequestError or whatever is raised might be preferable.

mdrxy committed:
let errors emit

mdrxy committed:
add docstring to tests for clarity

mdrxy committed:
fixes

mdrxy committed:
mock http request to prevent socket error

mdrxy committed:
fix prediction test

mdrxy committed:
fix

mdrxy committed:
add responses api verbosity TODO xfail test

mdrxy committed:
fix(openai): tests to content blocks (#32457)

Updates failing integration tests to use `output_version="responses/v1"`
and correctly access tool outputs from message content blocks instead of
additional_kwargs["tool_outputs"].

With `output_version="responses/v1"`, OpenAI's responses API stores tool
outputs directly in the message content array as structured blocks
rather than in `additional_kwargs["tool_outputs"]`.

The tests now:
1. Filter content blocks by type using `isinstance(block, dict)` and
`block.get("type") == "tool_type"`
2. Extract tool outputs from the filtered content blocks
3. Maintain the same assertion logic for validating tool output
structure

mdrxy committed:
simplify test
